### PR TITLE
fix: Use the `component` name source for SentryPerformanceTracker

### LINKED ISSUE #2125: Add transaction name source for SentryUIViewControllerPerformanceTracker transactions
### Description

When we start an automatic controller transaction, the name source being used is custom, we need to use component.

### PR DESCRIPTION
**Pull request by kevinrenskers**
## :scroll: Description

I don't think this needs a changelog entry, what do you guys think?
#skip-changelog

## :bulb: Motivation and Context

Closes #2125

## :green_heart: How did you test it?

Unit test

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

**Comment by brustolin** [comment]
> I don't think this needs a changelog entry, what do you guys think?

I think we do.

**Comment by kevinrenskers** [comment]
Ok, added

File: Sources/Sentry/SentryPerformanceTracker.m [hunk 1]
```diff
@@ -51,7 +51,9 @@ - (SentrySpanId *)startSpanWithName:(NSString *)name operation:(NSString *)opera
         newSpan = [activeSpan startChildWithOperation:operation description:name];
     } else {
         SentryTransactionContext *context =
-            [[SentryTransactionContext alloc] initWithName:name operation:operation];
+            [[SentryTransactionContext alloc] initWithName:name
+                                                nameSource:kSentryTransactionNameSourceComponent
```
Comment (brustolin) [inline Sources/Sentry/SentryPerformanceTracker.m:55]: SentryPerformanceTracker is a generic tool to create a transaction hierarchy, I don't think we should make this hard coded. 
I think this should be a parameter, and we could have an overload that uses component as default.

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**